### PR TITLE
Deprecated Cortex chunks storage

### PR DESCRIPTION
**Pull request by pracucci**
**What this PR does**:
During the last Cortex community meeting we discussed about deprecating the Cortex chunks storage and putting it into maintenance mode. This PR proposes to do the first step and announce it (CHANGELOG + doc).

No config change has been done to switch the default storage from chunks to blocks, in order to not introduce any breaking change.

I'm also proposing to delete `k8s/` manifests (as discussed during the last community meeting as well) because they're out of date, unmaintained and some config is also broken. We're seen some users getting confused by it so, unless we'll start actively maintaining and continuously testing them, I think it's better to get rid of them.

If this PR will be accepted & merged, I will also take care of gently closing all open PRs proposing non-bug fix changes to the chunks storage.

**Which issue(s) this PR fixes**:
N/A

**Checklist**
- [ ] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/configuration/arguments.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ Duration arguments should be specified with a unit like `5s` or `3h`. Valid time
 
 - `-querier.query-parallelism`
 
-   This refers to database queries against the store (e.g. Bigtable or DynamoDB).  This is the max subqueries run in parallel per higher-level query.
+   This refers to database queries against the store when running the Cortex chunks storage (e.g. Bigtable or DynamoDB).  This is the max subqueries run in parallel per higher-level query.
```
Comment (bboreham) [inline docs/configuration/arguments.md:23]:
```suggestion
   This refers to database queries against the store when running the deprecated Cortex chunks storage (e.g. Bigtable or DynamoDB).  This is the max subqueries run in parallel per higher-level query.
```

**Review by bboreham — APPROVED** [review]
Good job.  I think the engine will still default to chunks, so perhaps explain that, with reference to our compatibility guarantees.

**Comment by pracucci** [comment]
> I think the engine will still default to chunks, so perhaps explain that, with reference to our compatibility guarantees.

@bboreham Good point. Where do you suggest to mention it, other than the CHANGELOG?

**Comment by bboreham** [comment]
Perhaps at the top of `docs/configuration/arguments.md` ?

I think it could be on the flag too: https://github.com/cortexproject/cortex/blob/d382e1d80eaf0847e3b57592b970a4f1bf3121df/pkg/chunk/storage/factory.go#L114

E.g. "The storage engine to use: chunks (deprecated) or blocks."

**Comment by bboreham** [comment]
I guess we would also close all issues relating to the chunks storage: https://github.com/cortexproject/cortex/labels/storage%2Fchunks

**Review by jtlisi — APPROVED** [review]
LGTM

@pracucci Any thoughts on potential reducing the usage of chunks storage in the `e2e` tests on the repo? We are starting to hit close to 30 minutes per run and shaving a few minutes off would be nice.

**Comment by pstibrany** [comment]
Deprecating chunks sounds good to me, but having deprecated default value for non-deprecated option looks weird. Is it time to make Cortex 2 release to fix this, and do some options cleanup?

**Comment by bboreham** [comment]
Agreed on changing the default and on reducing e2e testing, later, but getting the "deprecated" message in front of people is a priority.

**Comment by pracucci** [comment]
I've created a meta issue to track the next steps about deprecating the chunks storage:
https://github.com/cortexproject/cortex/issues/4295

**Pull request merged**